### DISCUSSION TITLE
V2.4.13 build failure on ubuntu

### DISCUSSION THREAD

**Discussion by leisun124** [eProsima/Fast-DDS#5157] [General]
I have following build failure

  Building CXX object src/cpp/CMakeFiles/fastrtps.dir/rtps/network/NetworkFactory.cpp.o
In file included from /local/Fast-DDS-4/Fast-DDS/src/cpp/rtps/transport/TCPChannelResourceBasic.h:20:0,
                 from /local/Fast-DDS-4/Fast-DDS/src/cpp/rtps/transport/TCPAcceptorBasic.h:19,
                 from /local/Fast-DDS-4/Fast-DDS/src/cpp/rtps/transport/TCPTransportInterface.h:34,
                 from /local/Fast-DDS-4/Fast-DDS/src/cpp/rtps/network/NetworkFactory.cpp:27:
/local//Fast-DDS-4/Fast-DDS/src/cpp/rtps/transport/TCPChannelResource.h:244:45: error: wrong number of template arguments (1, should be 2)
             asio::basic_socket<asio::ip::tcp>& socket,
                                             ^
In file included from /usr/include/asio/basic_datagram_socket.hpp:20:0,
                 from /usr/include/asio.hpp:19,
                 from /local//Fast-DDS-4/Fast-DDS/src/cpp/rtps/transport/TCPTransportInterface.h:23,
                 from /local//Fast-DDS-4/Fast-DDS/src/cpp/rtps/network/NetworkFactory.cpp:27:
/usr/include/asio/basic_socket.hpp:41:7: note: provided for ‘template<class Protocol, class SocketService> class asio::basic_socket’
 class basic_socket

seems like it's not compatible with ASIO 1.10.8,  yet the cmake is finding packet for that version

eprosima_find_thirdparty(Asio asio VERSION 1.10.8)

Thanks
Chris

**Comment by leisun124** [comment]
correction, V2.14.3

**Reply by lucamorando95** [reply]
Did you find a solution? I have the same problem hee. It appears that the class definition is different now from the one accepting two argument

**Comment by MiguelCompany** [comment]
Which ubuntu version are you referring to?
Which process did you follow for building?

**Reply by Paraniod1** [reply]
The issue that appears when I use CMake for installation is shown above.
Ubuntu 18.04
https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html

**Comment by idesign0** [comment]
The issue is caused by the updated ASIO requiring two template parameters for basic_socket. Changing

asio::basic_socket<asio::ip::tcp>& socket

to

asio::basic_socket<asio::ip::tcp, asio::stream_socket_service<asio::ip::tcp>>& socket

fixes the build error for me (i needed it for ros2-fastDDS, using asio 1.10.8). Make sure to update all such occurrences in the code.

**Comment by MiguelCompany** [comment]
You could force Fast DDS to build against its asio submodule by adding `-DTHIRDPARTY_Asio=FORCE` to your cmake invocation
